llama-index-integrations/llms/llama-index-llms-bedrock-converse/llama_index/llms/bedrock_converse/base.py

```diff
@@ -179,9 +188,9 @@ def __init__(
         if client is not None:
             self._client = client
         elif "bedrock-runtime" in session.get_available_services():
-            self._client = session.client("bedrock-runtime", config=config)
+            self._client = session.client("bedrock-runtime", config=self._config)
         else:
-            self._client = session.client("bedrock", config=config)
+            self._client = session.client("bedrock", config=self._config)
```

Comment:
Curious why we keep `self._client` but for async we use the session and create a new client each time?

I don't know which approach is better, but would be nice if there was either a good reason to diverge or if they both worked the same 👀

Reply:
[As I mentioned in this comment](https://github.com/run-llama/llama_index/pull/14326/commits/c7595e7a54aab8d4e397fdf36ee579c71bde4061#diff-e2827a037be4cc0ee1f1b4910451e21f02a411a3eac88a222e21f456fd61f99bR308), the async client needs to be defined with an `async with`, otherwise I was getting an error. This error seems to be that the class wasn't being properly instantiated, as it didn't have the `converse` method.

Reply:
Would you prefer to use the same method for instantiating the sync client?

Reply:
nah its fine, that makes sense